### PR TITLE
web: migrate `useHistory` to `useNavigate`

### PR DESCRIPTION
**Pull request by valerybugakov**
## Context

- Part of #33834 
- With `Hide whitespace: true` the diff is much smaller 😜

## Test plan

CI

## App preview:

- [Web](https://sg-web-vb-react-router-migration-4.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (+0.04 kb) | -0.00% (-0.18 kb) | -0.00% (-0.21 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/vb/react-router-migration-4/compare-report.html;tab=live_object) for a full comparison between the commits 113f82d and ce38bbc or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

**Review by umpox — APPROVED** [review]

File: client/web/src/hooks/useScrollManager/useScrollManager.ts [hunk 1]
```diff
@@ -33,7 +34,8 @@ const SCROLL_RETRY_TIMEOUT = 3000
  * @param containerRef A React ref object of the container where scrolling will be managed.
  */
 export function useScrollManager(containerKey: string, containerRef: React.RefObject<HTMLElement>): void {
-    const { action, location } = useHistory()
+    const history = useHistory()
```
Comment (vovakulikov) [inline client/web/src/hooks/useScrollManager/useScrollManager.ts:37]: Will this "useHistory" migrate later?

Comment (vovakulikov) [reply client/web/src/hooks/useScrollManager/useScrollManager.ts:37]: Should we use globalHistory object for now?

**Review by vovakulikov — APPROVED** [review]
Left one comment about scroll manager hook

File: client/web/src/hooks/useScrollManager/useScrollManager.ts (hunk 1 above)
Comment (valerybugakov) [reply client/web/src/hooks/useScrollManager/useScrollManager.ts:37]: Yep, I'll remove it in a follow-up PR 👍

**Pull request merged**